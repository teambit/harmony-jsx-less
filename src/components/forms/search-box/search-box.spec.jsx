import React from 'react';
import testRenderer from 'react-test-renderer';
import {
  SearchBoxExample,
  SearchBoxWithPlaceHolder,
} from './search-box.composition';

describe('Search Box component', () => {
  it('renders correctly', () => {
    testRenderer.create(<SearchBoxExample />);
  });
  it('snapshot test', () => {
    const component = testRenderer.create(<SearchBoxExample />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with place holder props', () => {
    testRenderer.create(<SearchBoxWithPlaceHolder />);
  });
  it('snapshot test with place holder props', () => {
    const component = testRenderer.create(<SearchBoxWithPlaceHolder />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
