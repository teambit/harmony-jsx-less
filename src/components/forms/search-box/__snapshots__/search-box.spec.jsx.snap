// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box component snapshot test 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="sc-bdfBwQ jyVkcv"
    placeholder="Search..."
  />
  <button
    className="sc-gsTCUz fTGdlx"
    type="submit"
  >
    Search
  </button>
</form>
`;

exports[`Search Box component snapshot test with place holder props 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="sc-bdfBwQ jyVkcv"
    placeholder="place holder"
  />
  <button
    className="sc-gsTCUz fTGdlx"
    type="submit"
  >
    Search
  </button>
</form>
`;
